inc/tokens/class-authorization-code.php

```diff
@@ -108,6 +108,36 @@ public function get_expiration() {
 		return (int) $value['expiration'];
 	}
 
+	private function validate_code_verifier( $args ) {
+		$value = $this->get_value();
+
+		if ( empty( $value['code_challenge'] ) ) {
+			return true;
+		}
+
+		$code_verifier = $args['code_verifier'];
+		$is_valid = false;
+
+		switch ( strtolower( $value['code_challenge_method'] ) ) {
+			case 's256':
+				$decoded = base64_encode( hash( 'sha256', $code_verifier ) );
+				$is_valid = $decoded === $value['code_challenge'];
+				break;
+			case 'plain':
+				$is_valid = $code_verifier === $value['code_challenge'];
+			break;
```

Comment:
We should add a `default` case for unsupported algorithms; [§4.4.1](https://tools.ietf.org/html/rfc7636#section-4.4.1) states that the description `SHOULD` disambiguate this case.

Reply:
"If the server supporting PKCE does not support the requested
   transformation, the authorization endpoint MUST return the
   authorization error response with "error" value set to
   "invalid_request"" - ```inc/types/class-base/php``` line 192 - the ```challenge_method``` cannot be anything else than _plain_ | _s256_. However, I edited the error message according to the standard and added a default throwing an exception nonetheless.